### PR TITLE
support build from local source code

### PR DESCRIPTION
**Pull request by wanjunlei**
Signed-off-by: wanjunlei <wanjunlei@kubesphere.io>

### CODE REVIEW

File: apis/core/v1beta1/function_webhook.go [hunk 1]
```diff
@@ -167,9 +167,14 @@ func (r *Function) ValidateBuild() error {
...
 
-	if r.Spec.Build.SrcRepo != nil && r.Spec.Build.SrcRepo.Url == "" {
-		return field.Required(field.NewPath("spec", "build", "srcRepo", "url"),
-			"must be specified when `spec.build.srcRepo` enabled")
+	if r.Spec.Build.SrcRepo == nil {
+		return field.Required(field.NewPath("spec", "build", "srcRepo"),
+			"must be specified when `spec.build` enabled")
+	}
+
+	if r.Spec.Build.SrcRepo.Url == "" && r.Spec.Build.SrcRepo.BundleContainer == nil {
```
Comment (benjaminhuo) [inline apis/core/v1beta1/function_webhook.go:175]: Maybe we need to add a check to ensure only one of the URL or bundleContainer can be specified?
Not all of them can be set simultaneously.

Comment (wanjunlei) [reply apis/core/v1beta1/function_webhook.go:175]: If `bundleContainer` and `url` are set at the same time, `bundleContainer` will be used first.

Comment (benjaminhuo) [reply apis/core/v1beta1/function_webhook.go:175]: ok

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
We need docs, helm charts for the revision-controller now.
Are there any other changes needed before releasing v1.0?

**Pull request merged**